### ISSUE TITLE
Zimbra folder view not visible with PrivacyBadger installed

### ISSUE THREAD

**Issue by ufkub** [EFForg/privacybadger#1050]
In the Zimbra Webmail client, the folders on the left side are not visible anymore when PrivacyBadger is active.

This is also visible on a Zimbra demo account like https://www.grmail.ca/en/demo.
See attached screenshots.

* PrivacyBadger installed, but deactivated on the Firefox Extension page: OK. Folders are shown. 
* PrivacyBadger installed and activated: Bad. Folders not visible.
* PrivacyBadger installed and activated, but deactivated for the current Zimbra site: Bad. Folders not visible. Deactivating PrivacyBadger only for this site dies not make a difference.

Occurs since the PrivacyBadger update to 2016.12.08 today.

Firefox: 50.0.2+linuxmint1+serena
PrivacyBadger: 2016.12.08

![zimbra-demo_privacybadger_activated](https://cloud.githubusercontent.com/assets/12187056/21149439/c6e863c4-c15b-11e6-906d-346444d9d8cb.png)
![zimbra-demo_privacybadger_deactivated](https://cloud.githubusercontent.com/assets/12187056/21149440/c70b452e-c15b-11e6-8e11-4b376c982a26.png)
![Uploading zimbra-demo_privacybadger_activated.png…]()
![Uploading zimbra-demo_privacybadger_deactivated.png…]()

**Comment by ufkub** [comment]
Additionally, the following warnings are written to the command line. Is all that from PrivacyBadger?

1481646585491	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Error processing permissions.9: Unknown permission "privacy"
1481646585495	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Error processing author: An unexpected property was found in the WebExtension manifest.
1481646585495	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Error processing minimum_chrome_version: An unexpected property was found in the WebExtension manifest.
1481646585496	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Error processing options_page: An unexpected property was found in the WebExtension manifest.
1481646585496	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Error processing update_url: An unexpected property was found in the WebExtension manifest.
1481646585541	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Please specify whether you want browser_style or not in your browser_action options.
Extension error: TypeError: this.tabData[tabId] is undefined moz-extension://13313db2-f444-4032-a0d3-78bbc42e51a9/src/background.js 405
[[Exception stack
Badger.prototype.getAllOriginsForTab@moz-extension://13313db2-f444-4032-a0d3-78bbc42e51a9/src/background.js:405:5
refreshPopup@moz-extension://13313db2-f444-4032-a0d3-78bbc42e51a9/src/popup.js:258:17
@moz-extension://13313db2-f444-4032-a0d3-78bbc42e51a9/src/popup.js:540:5
getTab/<@moz-extension://13313db2-f444-4032-a0d3-78bbc42e51a9/src/popup.js:534:76
Current stack
runSafeSyncWithoutClone@resource://gre/modules/ExtensionUtils.jsm:43:129
runSafeSync@resource://gre/modules/ExtensionUtils.jsm:74:37
runSafe@resource://gre/modules/ExtensionUtils.jsm:220:32
wrapPromise/<@resource://gre/modules/ExtensionUtils.jsm:399:13

**Comment by groovecoder** [comment]
With PB enabled, the site also throws this error:
```
TypeError: ERROR.captureStackTrace is not a function   zimbra.gr-web.biz:81:7
```

**Comment by groovecoder** [comment]
`ERROR.captureStackTrace` is a v8-only method. [Babel has a solid polyfill](https://github.com/babel/babel/commit/719d23eb3b94598a0ba41f339b52d54d2395b99f), and there's an [error-polyfill](https://www.npmjs.com/package/error-polyfill) npm module.

But this error is being thrown by the zimbra.gr-web.biz resource? Still digging to see if it's Zimbra's problem or PB's problem.

**Comment by groovecoder** [comment]
Filed a PR that fixes the `fingerprinting.js` content script from breaking Firefox, but it doesn't implement the full `captureStackTrace` functionality. That may be a follow-up issue?

**Comment by ufkub** [comment]
> Still digging to see if it's Zimbra's problem or PB's problem.

The bug is not related to Zimbra. I see a similar problem on http://en.chessbase.com/post/yasser-seirawan-on-magnus-carlsen-vs-sergey-karjakin_3.

* PrivacyBadger activated: Board window empty.
* PrivacyBadger deactivated: There's the expected chessboard.

![chessgame_privacybadger_activated](https://cloud.githubusercontent.com/assets/12187056/21160846/a96f9396-c186-11e6-88a2-52e913495a2f.png)
![chessgame_privacybadger_deactivated](https://cloud.githubusercontent.com/assets/12187056/21160847/a991761e-c186-11e6-97fc-9b610bcbc652.png)

**Comment by ianmcorvidae** [comment]
I'm having the same issue on https://use.jetstream-cloud.org/application. To be very specific, it's failing in this case when Modernizr feature detection attempts to call the `toDataURL` function of a `canvas` element.

To reproduce specifically this, go to data:text/html,<html></html> with and without PB, and use the console to try to run `document.createElement("canvas").toDataURL("image/jpeg")`. Enabled, it'll throw the `captureStackTrace` error; disabled it'll work fine.

The monkey-patch may still be the appropriate fix, just wanted to bring in a very narrow reproducible case in case that's helpful here.

**Comment by cooperq** [comment]
This should be fixed by version 2016.12.15.1 please re-open if not.

**Comment by ufkub** [comment]
Yes, that fixed it. Thank you.

**Comment by KnuckleBob** [comment]
This issue has re-emerged as part of Privacy Badger 2017.9.12. The folder view in Zimbra is no longer visible.

**Comment by ghostwords** [comment]
This should be fixed again as of Privacy Badger 2017.9.12.1.
